core/templates/dev/head/pages/exploration_editor/simple_editor_tab/QuestionHashService.js

```diff
@@ -0,0 +1,57 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that contains common functionalities for creating
+unique Question identifiers.
+ */
+
+oppia.factory('QuestionHashService', [function() {
+  return {
+    getHash: function(question, subfield) {
+      var questionId;
+      var suffix;
+      if (angular.isDefined(question)) {
+        if (angular.isString(question)) {
+          questionId = question.toLowerCase().replace(' ', '-');
+        } else if (angular.isObject(question)) {
+          questionId = question.getStateName().toLowerCase().replace(' ', '-');
+        }
+      };
+      if (angular.isDefined(subfield)) {
+        if (angular.isString(subfield)) {
+          suffix = subfield;
+        } else if (angular.isObject(subfield)) {
+          suffix = subfield.abbr;
+        }
+      };
+      if (angular.isDefined(suffix)) {
+        return [questionId, suffix].join('-');
+      } else {
+        return questionId;
+      };
+    },
+    getSidebarItemHash: function(question, subfield) {
+      return 'si-' + this.getHash(question, subfield);
```

Comment:
No harm using longer hashes like 'sidebaritem-'. That way, it's clear what they mean. Otherwise it's hard for the developer to keep the map of abbreviations in their head when debugging.